enhancements/removing-midstream.md

```diff
@@ -0,0 +1,322 @@
+Pipelines, it is not necessary anymore. The aim is to remove it and
+have this enhancement proposal describing what we propose to do.
+
+## Motivation
+
+As said above, the midstream repository are now more work than they
+are worth. They also have CI attached but that doesn't reflect the
+product and thus CI results is most not useful.
+
+## Goals
+
+- Remove the use of midstream repositories (any `openshift/tekton-` repositories)
+- Document what is the new "release" workflow from upstream to
+  downstream
+- Explore possible alternatives
+
+## Requirements
+
+- It should be possible to apply "out-of-upstream" patches on the
+  source code before building the product.
+- Ideally those set of patches are publicly available (for
+  transparency)
+- Re-use as much as we can from upstream and/or downstream tooling
+- The process should be automated as much as possible
+
+## Proposal
+
+There is two different case to handle:
+- Components (pipeline, triggers, chains, …)
+- Operator
+
+Let's dig into what needs to be done for both here, and then we can try to define a set of
+things to run (in Task/Pipeline) to handle both these cases.
+
+### Components
+
+For each components we ship as part of OpenShift Pipelines, there is two simple steps to do:
+- Fetch the upstream repository at a given branch
+- Apply patches if need be (if they didn't get in, …)
+- Build the images
+
+
+### Operator
+
+For the operator, it is a bit trickier as we have an extra step to take:
+- Fetch the upstream repository at a given branch
+- Fetch the component's payloads at a given release / nightly
+- Apply patches if need be (same as above, …)
+- Build the images
+
+### Combination of the two
+
+If we combine both, it looks a little bit like the following
+
+- Fetch the upstream repository
+  - Get the latest commit of the tracked branch (or tag)
+  - Update `upstream_sources.yml` with it
+- Read the operator metadata and fetch payload version
+  - latest means we take nightly payload
+  - otherwise we get the released payload
+- Apply patches if need be
+- Build images
+  - ahead of time to not submit a changeset if it doesn't build
+  - or just send a changeset
+
+This two pieces can happen when we track new commits in the branches we track. As of today, CPaaS
+doesn't seem to support this kind of custom behavior inside the "poll upstream job", but we could
+deactivate it and have our own, living in our long-living *dogfooding* cluster.
+
+In addition, it should be possible to make this all work locally on your laptop. What this means
+is that it should be possible to update upstream source from upstream repositories and this would
+allow you to get the proper payload apply patches and build the images.
+
+## Design detail
+
+The idea is to replace the CPaaS `poll-ustream` job with our own "process", and in our case, our 
+own Tekton Pipeline. So the first step is to de-activate the automated `poll-upstream` Job.
```

Comment:
I love this idea. The best part is that the logical context/intention of the new "our own process" can be tailored to fit our team's needs. Mechanisms like CPaaS poll-upstream or any script we write upstream is lean towards being generic which makes it difficult to automate/update content for a Productized release.

also typo `poll-ustream`🤓